### PR TITLE
Jbpm Performance Tests

### PR DESCRIPTION
**Pull request by ibek**
Basic performance tests for jbpm engine. Remote API tests will follow when the execution servers are unified.

jbpm-performance-tests: added runtimeManagerStrategy
jbpm-performance-tests: refactored constants and added LGroupHumanTaskProcess
jbpm-performance-tests: Added L1000HumanTasksQueryPagination
jbpm-performance-tests: Added support for pessimistic and optimistic locking
jbpm-performance-tests: Added Parallel gateway tests and RuleTaskTest
jbpm-performance-tests: Added perf tests for signal, timer, script and startend
jbpm-performance-tests: Added L1000ProcessesSignal
jbpm-performance-tests: fixed metrics to report to right csv files
jbpm-performance-tests: Added JBPMTestConfig and limited L1000HT scenarios
jbpm-performance-tests: Format report output

### CODE REVIEW

**Comment by ibek** [comment]
The performance module and this pull request depends on KIE Performance KIT from https://github.com/droolsjbpm/droolsjbpm-integration/pull/143

**Comment by mswiderski** [comment]
Ivo, do you mind add some readme that will inform users (and us) how to use this module.

File: performance-tests/jbpm-performance-tests/pom.xml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    </dependency>
+    <dependency>
+      <groupId>io.dropwizard.metrics</groupId>
+      <artifactId>metrics-core</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.dropwizard.metrics</groupId>
+      <artifactId>metrics-jvm</artifactId>
+    </dependency>
+    <dependency>
```
Comment (mrietveld) [inline performance-tests/jbpm-performance-tests/pom.xml]: Dropwizard! Huh, interesting, I didn't know they had a metrics library. I'm curious to see how you use that and how it works.

**Comment by mrietveld** [comment]
@ibek This looks great! I believe you have committer rights: please go ahead and merge this when you want to.

**Comment by ibek** [comment]
Thanks Marco

**Pull request merged**